### ISSUE TITLE
Expose voice dialog capabilities

### ISSUE THREAD

**Issue by GiviMAD** [openhab/openhab-core#2688]
Hi, I have been developing two add-ons a [Speech-to-Text service](https://github.com/openhab/openhab-addons/pull/12055) and a [Keyword Spotter service](https://github.com/openhab/openhab-addons/pull/12028) so I can add some voice capabilities to my current openHAB installation. Both seems to work fine with the build-in dialog processor and I can get text on the rule processor.

The problem is that for enabling this I have to inject a custom service that calls the startDialog method on the VoiceManager, as this capability is not exposed right now. (Also the interface do not expose the capability to stop the dialog processing).

Could this be exposed through the voice service configuration on the 3.3.0-Snapshot branch? This way those add-ons could be easily tested by other users.

Thank you in advance.

**Comment by lolodomo** [comment]
By exposing, you mean an action that could be called on a rule? 
You're right, this is missing. 

startDialog() starts an infinite dialog or just a one shot dialog keyword => STT => HLI => TTS? 

What you are adding (keyword service + STT service) looks interesting, especially  when we still have nothing until today.

**Comment by lolodomo** [comment]
And the voice REST API should be enhanced too.

**Comment by GiviMAD** [comment]
```
startDialog() starts an infinite dialog or just a one shot dialog keyword => STT => HLI => TTS?
```
An infinite loop between Keyword => STT => HLI. TTS is not necessary used on each iteration, it is used to inform from errors on the process or from the HLI depending on how it is implemented.
```
By exposing, you mean an action that could be called in a rule?
You're right, this is missing.
```
For my the ideal solution is adding a boolean to the audio configuration like 'Enable dialog' that allows to control whether this capability is enabled. This could be a little tricky as the Audio Manager will need to wait for other services to be activated on start up to start the dialog processing. I don't know how hard is to achieve this.

Having it accesible on rules will be also nice as will allow to do more complex things, as changing the audio stream we are processing, but I think from rules a method for a single execution STT => HLI will be more useful, as probably if you are using it from rules you what to bypass the wake word. This probably could be left for the future, but seems like a good idea for performing remote audio processing or things like that.

```
And through a console command, missing too.
```
Yes, commands to stop and start this capability will be nice to have also, maybe we could start by those.

Let my know if you want me to help on any of this, will be nice to learn more things about the core.

**Comment by lolodomo** [comment]
Ok, what you expect is an option for an automatic start of the default dialog processing. 
Note that dialog processing is manages by the voice manager, not the audio manager. 

I have a relative clear idea of what to change. I will expose my ideas later today and I could even implement the first part very quickly (as it will be very simple).

**Comment by GiviMAD** [comment]
Sorry I switch all the time the name of these services. I meant the Voice Manager, I've edited it.

**Comment by lolodomo** [comment]
Here is my proposal:

1. (breaking change) Extend the VoiceManager interface to add a method `stopDialog` with a nullable parameter source.
2. Update the voice manager implementation to allow only one running dialog per audio source + coding of the new method `stopDialog` .
3. Add 2 new console commands: `voice startdialog [<source> <hli> <sink>]` and `voice stopdialog [<source>]`
4. Add a new voice configuration parameter to enable or disable the automatic start of the default dialog
5. Enhance the voice REST API to allow starting/stopping a dialog
6. Add new actions to allow starting/stopping a dialog from a rule

I plan to create a first PR covering at least items 1 2 3.

Is it ok for you ?

**Comment by GiviMAD** [comment]
Yes, I didn't realize we should link the dialog execution to the source in order to be able to stop them correctly.
Every thing looks good for me, thanks for the quick response.

**Comment by GiviMAD** [comment]
I just realice maybe in the point 3 the start dialog command should also accept the keyword spotter service to use as a param.

**Comment by lolodomo** [comment]
@GiviMAD : to be clarified who should take care closing the audio source, either DialogProcessor or each KS/STT service.

**Comment by lolodomo** [comment]
> I just realize maybe in the point 3 the start dialog command should also accept the keyword spotter to use as a param.

Ok, I will add it,

**Comment by lolodomo** [comment]
To be fixed in dialog processing: few messages hardcoded in English used with TTS.

**Comment by lolodomo** [comment]
Not sure this is currently done.

**Comment by GiviMAD** [comment]
> Not sure this is currently done.

With the current implementation I have no need to open or close the audio streams and everything works. Maybe I miss understood the question.

**Comment by lolodomo** [comment]
I mean the dialog processor gets an input stream ( `source.getInputStream(fmt)` ) when calling `ks.spot` and `stt.recognize`.
The `getInputStream` call opens the input stream. But I see no call to close this stream in the code of the dialog processor.
Of course, it works but I assume some resources are not released properly.

**Comment by lolodomo** [comment]
Keyword parameter added for the console command.

**Comment by GiviMAD** [comment]
> I mean the dialog processor gets an input stream ( `source.getInputStream(fmt)` ) when calling `ks.spot` and `stt.recognize`.
> The `getInputStream` call opens the input stream. But I see no call to close this stream in the code of the dialog processor.
> Of course, it works but I assume some resources are not released properly.

Thanks for the clarification, I think that will be easier if the Dialog processor is on charge of this, as far I saw the stream passed to the ks should be only closed when the associated dialog is stopped, but for the one passed to the stt it could be closed after each execution. Other way we always need to close the stream stream inside the abort method of the returned instance which seems easy to forget.

Watching this I saw I need to do a small change on the googlestt add-on, as I didn't know that a SpeechRecognitionErrorEvent aborts the service execution.

**Comment by lolodomo** [comment]
I implemented the automatic start of the default dialog processor. But I believe it can fail it the default audio source or link are loaded  very late in the audio manager.
I try to start each time a new KS/STT/HLI/TTS service is loaded. But the audio sources/sinks are managed by the audio manager, not the voice manager, so I can't do the same thing (my code is inside the voice manager).
An alternative would be to have a scheduled task that will try every XX seconds until success. But having a scheduled task just for that...

**Comment by GiviMAD** [comment]
> I implemented the automatic start of the default dialog processor. But I believe it can fail it the default audio source or link are loaded very late in the audio manager.
> I try to start each time a new KS/STT/HLI/TTS service is loaded. But the audio sources/sinks are managed by the audio manager, not the voice manager, so I can't do the same thing (my code is inside the voice manager).
> An alternative would be to have a scheduled task that will try every XX seconds until success. But having a scheduled task just for that...

Ummm. Maybe we could make some kid of link between voice manager and the audio manager? Something like the audio manager requiring the voice manager and notifying it when sources are ready? Maybe is difficult to make but seems as will be the way to make this right as the voice manager really has a dependency on audio manager.

**Comment by lolodomo** [comment]
I am going to suppress the auto start option as I am not fully satisfied with my implementation.
Then I will try to enhance the integration tests.

I am not able to test a dialog myself.
Will you be able to test my changes on your side (before the merge) ? To be tested: no regression + new console commands to start and stop a dialog.

**Comment by GiviMAD** [comment]
Sure, I'll take a look this evening.

**Comment by lolodomo** [comment]
> Maybe we could make some of link between voice manager and the audio manager? Something like the audio manager requiring the voice manager and notifying it when sources are ready? Maybe is difficult to make but seems as will be the way to make this right as the voice manager really has a dependency on audio manager.

A link is already existing, the voice manager references the audio manager.
The "problem" is that audio sources and sinks are loaded/unloaded dynamically by the audio manager.
When the voice manager starts, you know the audio manager is already started but you cannot be sure whether a particular audio source or audio sink is already loaded.
That is the same for a STT/TTS/HLI/KS service in the voice manager, except that here the voice manager knows when such services are loaded/unloaded (because they are "referenced").
We could "reference" the audio sources/sinks in the voice manager but this will be more or less a bypass of the audio manager.
Remember the principle of openHAB is OSGi bundles that can be loaded/unloaded at any time.
To conclude, I have no clear idea how to implement your use case.

Normally, we should even stop any dialog processor as soon as one of the used services is unloaded. This is not managed.

**Comment by GiviMAD** [comment]
> > Maybe we could make some of link between voice manager and the audio manager? Something like the audio manager requiring the voice manager and notifying it when sources are ready? Maybe is difficult to make but seems as will be the way to make this right as the voice manager really has a dependency on audio manager.
> 
> A link is already existing, the voice manager references the audio manager. The "problem" is that audio sources and sinks are loaded/unloaded dynamically by the audio manager. When the voice manager starts, you know the audio manager is already started but you cannot be sure whether a particular audio source or audio sink is already loaded. That is the same for a STT/TTS/HLI/KS service in the voice manager, except that here the voice manager knows when such services are loaded/unloaded (because they are "referenced"). We could "reference" the audio sources/sinks in the voice manager but this will be more or less a bypass of the audio manager. Remember the principle of openHAB is OSGi bundles that can be loaded/unloaded at any time. To conclude, I have no clear idea how to implement your use case.
> 
> Normally, we should even stop any dialog processor as soon as one of the used services is unloaded. This is not managed.

Maybe the Audio manager should emit events when the audio sink/sources are loaded unloaded, and also allow other services to ask for their status so they can decide if every thing is ready to start or they should wait for an event to be emitted on the sink/sources their are going to use. Anyway I think that this pr covers enough to start testing things maybe at the end we can create another issue describing the current problem in more detail.

I was doing some testing in your branch and the commands works correctly, but there is something broken in the voice manager, because first dialog execution works fine, but the second one the stream passed to the STT service always return 0, and ends causing and out of memory. That didn't happen with the previous implementation. 
I will try to figure out what is happening tomorrow, I didn't have much time today. Let me know if you find something.

**Comment by GiviMAD** [comment]
Take a look at the JavaSoundAudioSource implementation bundles/org.openhab.core.audio/src/main/java/org/openhab/core/audio/internal/javasound/JavaSoundAudioSource.java, method getInputStream.
First time you ask for the stream it inits the mic passing it to the method initMicrophone, then they store a reference to the returned object and return it to you wrapped in a JavaSoundInputStream, if you look at its implementation the mic (called input there is closed) when you call the close method. So next executions return a mic of getInputStream use the previous mic which has been closed previously when calling close on the previous input stream.
These is the code I'm talking about:

 JavaSoundAudioSource.java:
```
        TargetDataLine mic = this.microphone;
        if (mic == null) {
            mic = initMicrophone(format);
        }
        return new JavaSoundInputStream(mic, audioFormat);
```

JavaSoundInputStream.java
```
    @Override
    public void close() throws IOException {
        input.close();
    }
```

**Comment by lolodomo** [comment]
As you already guessed, the problem is certainly due to the closing of input stream I added in the dialog processor, in particular the one for STT..

Is the problem in the javasound or in the updated dialog processor, I don't know yet. I understand that you think it is in the javasound.

There is something I do not understand how it can work. We are opening 2 audio streams on the microphone, one for KS and one for STT. The one for STT is opened only while extracting the question after detecting the keyword, while the one for keyword spotting is opened permanently. How can 2 applications have the control on the microphone at the same time ?

I have now enhanced the integration tests, a full sequence KS => STT => HLI => TTS is now tested. They could still be improved to test the different error cases.

Other question for you: I see a lot of different events but in fact only a part of really considered by the dialog processor. Are all the events clear for you, for example for STT: SpeechStartEvent, SpeechStopEvent, RecognitionStartEvent, RecognitionStopEvent

And last question; is it clear for you when using KSException and when using KSErrorEvent ?

**Comment by lolodomo** [comment]
Reading again your message, I understand now the problem with close. 
Maybe there was a reason to not call everytime initMicrophone(format) ?

**Comment by lolodomo** [comment]
I could comment my code closing the input stream, adding a clear explanation why. WDYT? 

Regarding the stopDialig not working for you, do you get an IllegalStateException? If not, this method just calls the abort of you STT service (if not yet aborted( and KS service.

**Comment by GiviMAD** [comment]
> Reading again your message, I understand now the problem with close.
> Maybe there was a reason to not call everytime initMicrophone(format) ?

Even if there is a reason to do that (maybe there is a limitation in some environment), something is wrong there because if they are going to share an instance they can not close it when you close a single stream using it, it should not be shared or be implemented in a way to only close the mic when no more streams depend on it and ensuring it will be correctly initialised again next time.

So as a summary I think we should check with the autor if is something he did based on an issue or just trying to optimise it, and if so undo it and wait for users feedback before implementing something more complex.

> Regarding the stopDialig not working for you, do you get an IllegalStateException? If not, this method just calls the abort of your STT service (if not yet aborted( and of your KS service.

I didn't get any logs when running the command so no idea what is happening, I will check it in deep this afternoon.

**Comment by lolodomo** [comment]
> Even if there is a reason to do that (maybe there is a limitation in some environment), something is wrong there because if they are going to share an instance they can not close it when you close a single stream using it, it should not be shared or be implemented in a way to only close the mic when no more streams depend on it and ensuring it will be correctly initialised again next time.

Of course I agree with you on that point.

**Comment by GiviMAD** [comment]
Sorry I totally missed a message and let some things without answer. I will start for the message I missed.
> There is something I do not understand how it can work. We are opening 2 audio streams on the microphone, one for KS and one for STT. The one for STT is opened only while extracting the question after detecting the keyword, while the one for keyword spotting is opened permanently. How can 2 applications have the control on the microphone at the same time ?

That's something I don't understand either. Have not sense for me and more when they are sharing the same input reference.

> I see a lot of different events but in fact only a part of really considered by the dialog processor. Are all the events clear for you, for example for STT: SpeechStartEvent, SpeechStopEvent, RecognitionStartEvent, RecognitionStopEvent
AudioStartEvent AudioStopEvent

Never have to use these four: RecognitionStartEvent, RecognitionStopEvent,AudioStartEvent AudioStopEvent.
What was unclear to me at the beginning was that I have to emit a SpeechRecognitionErrorEvent when no transcription happens. Maybe there should be a dedicated kind of exception event for that because I think could be quite common and should not be handled as other errors.

> And last question; is it clear for you when using KSException and when using KSErrorEvent ?

I think so KSException for synchronous errors when invoking the spot method (example: library not loading) and KSErrorEvent for the async ones.

> Maybe a DEBUG log should also be added in your KS)STT abort methods?

Sure, now that I have the commands to start/stop dialog I will test that part and add more logs (once I discover why the stop fails for me). I'll start by the comments you left on the ks service pr, I think I can have it all done between today and tomorrow (hopefully).

**Comment by lolodomo** [comment]
I have to revert my small changes about configuration handling.

**Comment by GiviMAD** [comment]
I have been testing things with the change I told you in the JavaSoundAudioSource class:
```
        TargetDataLine mic = initMicrophone(format);
```
For me, the recognition got improved since I did that change. I think I was losing some parts of the audio due to race errors, because some times I got strange transcriptions or words got missed.
I was thinking it was due to the service was not so good for Spanish but since I applied that change those problems disappear.

Is it better to open a separate issue for this problem? In my opinion is better to have it solved before merging this one, so the streams can be closed as you do right now. What do you think?

**Comment by lolodomo** [comment]
Yes please open a separate issue and submit a PR with your proposed fix,

**Comment by lolodomo** [comment]
PR updated:

- Closing stream is commented (with a comment to explain why)
- DEBUG log added when starting a new dialog and when a dialog is stopped
- Revert my updated handling of configuration settings

**Comment by lolodomo** [comment]
Integration tests finished.

What remains to do is to replace hardcoded  texts that are said when something is going wrong by texts that can be translated in any language.

**Comment by lolodomo** [comment]
@GiviMAD : I consider now my PR as ready to be merged. We have to wait for its review. Once merged, I will continue with REST and actions.

**Comment by GiviMAD** [comment]
Hey @lolodomo, one comment on the current dialog processor implementation.
Is about [this](https://github.com/openhab/openhab-core/blob/8c16cb6e932d3fd4f2b967b6bf06dca62ff523a6/bundles/org.openhab.core.voice/src/main/java/org/openhab/core/voice/internal/DialogProcessor.java#L212) call to toggleProcessing.
I think this call should be moved to the stt start event, so everything is ready inside the add-on and is really listen for the input.

**Comment by lolodomo** [comment]
First off topic: @GiviMAD is it possible to use a microphone connected to a remote server? I am owning an old PS3 Eye. Can you confirm that the solution would be to connect it to a remote RPI having pulseaudio installed and then use the pulseaudio binding in my OH server to create an audio source in OH with this remote microphone?

**Comment by lolodomo** [comment]
Second off topic : regarding the interpreter, I started something 2 years ago to use HABot as an interpreter. The conclusion was that my approach was not good in terms of architecture, and I stopped my effort. The good approach would be to create a new binding based on openNLP and reusing w large part of what us in HABot. The challenge is mainly building decent sentences as result of interpretation. Maybe U could restart this project.

**Comment by GiviMAD** [comment]
> First off topic: @GiviMAD is it possible to use a microphone connected to a remote server? I am owning an old PS3 Eye. Can you confirm that the solution would be to connect it to a remote RPI having pulseaudio installed and then use the pulseaudio binding in my OH server to create an audio source in OH with this remote microphone?

I think that the pulse audio binding is not ready for this, but I didn't look to much at it, but I think the audio source part is still not ready

**Comment by GiviMAD** [comment]
> Second off topic : regarding the interpreter, I started something 2 years ago to use HABot as an interpreter. The conclusion was that my approach was not good in terms of architecture, and I stopped my effort. The good approach would be to create a new binding based on openNLP and reusing w large part of what is in HABot. The challenge is mainly building decent sentences as result of interpretation. Maybe I could restart this project.

Thats funny @lolodomo, I'm actually working on an openNLP interpreter, I started yesterday. I will open a WIP PR in the afternoon so you can see how it goes, for now I have done the processing part, but I'm woking on how to propagate the results to items.

**Comment by lolodomo** [comment]
> Thats funny @lolodomo, I'm actually working on an openNLP interpreter, I started yesterday. I will open a WIP PR in the afternoon so you can see how it goes, for now I have done the processing part, but I'm woking on how to propagate the results to items.

Propagating the result to items is not the job of a HLI binding. You just have to implement HLI interface and so return a text. 
Like HABot, the interpreter should rely on semantic tags.

**Comment by lolodomo** [comment]
> I think that the pulse audio binding is not ready for this, but I didn't look to much at it, but I think the audio source part is still not ready

You are using a microphone directly connected to your OH server?

**Comment by lolodomo** [comment]
@GiviMAD : now I understand what you mean by "propagating to items". Of  course, the HLI must send commands to items.

**Comment by lolodomo** [comment]
Maybe you can start from scratch but take a look to HABot, I am sure there is part of code code to be reused.

**Comment by lolodomo** [comment]
And as a first step, you could consider all intents covered by the standard interpreter packaged with OH. And then new intents could be added later.

**Comment by lolodomo** [comment]
@GiviMAD : can you please start a new "issue" to discuss about openNLP? I know I started the off topic ;) 
This issue will be closed when start/stop dialog processing are implemented as REST API and as rule actions.

**Comment by GiviMAD** [comment]
I'm already working on an idea. Please wait until I finish to see if it accomplish to you.
We can discuss the changes over the implementation or improvements in the PR when I get something functional.

**Comment by lolodomo** [comment]
@GiviMAD : as the new REST API will be available probably tomorrow (PR was merged), please make a try and tell if you see any problem.

I have implemented the rule actions but I still need to test.

**Comment by GiviMAD** [comment]
Sure I will take a look in the afternoon.

**Comment by GiviMAD** [comment]
I think that something is missing in the whole dialog implementation is the ability to start a dialog without user interaction. 
I was thinking on two useful method:

* one that behave like if the a spot event was fired (This could be useful if the user wants to create a button on the interface to trigger the dialog).
* another that allows to trigger the configured STT service and get the transcribed text (this one could be useful for asking for conformation from rules for example).
 What do you think? Is it better if I create a separate issue for those? Or do you think is something that doesn't need to be added?

**Comment by lolodomo** [comment]
> @lolodomo thanks for pinging me I didn't know about the ActionServices, I think is want I need to expose some actions from the HLI I'm building.

You have already several "interpret" actions. Are they not sufficient ?! What would you like to add ?

> Also I think that something is missing in the whole dialog implementation is the ability to start a dialog without user interaction. I was thinking on two useful method:
> 
>     * one that behave like if the a spot event was fired (This could be useful if the user wants to create a button on the interface to trigger the dialog).

I think we could add a new method named "listenNowAndAnswer" without the keyword spotter parameter and without the keyword parameter. This will not be an infinite dialog but rather a one shot question/answer dialog without keyword spotting.

>     * another that allows to trigger the configured STT service and get the transcribed text (this one could be useful for asking for conformation from rules for example).

We already have several actions for HLI and TTS, but none for STT. So why not.

>       What do you think? Is it better if I create a separate issue for those? Or do you think is something that doesn't need to be added?

The last 2 ones look interesting to me.

For the missing TTS actions, yes please open a new issue to expose what new actions you would like to add.

For the STT actions, yes please open a new issue to discuss what could be these actions.

For the method "listenNowAndAnswer", I propose to continue in this issue. Are you ok with this name ? I could add the method with the corresponding REST API and the new action after my last current PR is merged if you like.

**Comment by GiviMAD** [comment]
> You have already several "interpret" actions. Are they not sufficient ?! What would you like to add ?

I think I explained myself wrong, I was talking on adding some actions on the pr I was working on the add-ons repo, I want to expose some text analysis capabilities from the add-on to the rules but I think I'm leaving that for now. Don't worry, ignore that comment, is nothing related to the core, so sorry for mixing things up.

> For the method "listenNowAndAnswer", I propose to continue in this issue. Are you ok with this name ? I could add the method with the corresponding REST API and the new action after my last current PR is merged if you like.

Yes, the name sounds good to me. And of course, when you have time that would be great.

> For the STT actions, yes please open a new issue to discuss what could be these actions.

Ok I will open another issue for this one, thank you!

**Comment by lolodomo** [comment]
> > For the method "listenNowAndAnswer", I propose to continue in this issue. Are you ok with this name ? I could add the method with the corresponding REST API and the new action after my last current PR is merged if you like.
> 
> Yes, the name sounds good to me. And of course, when you have time that would be great.

Maybe listenAndInterpret or recognizeAndInterpret  would be better ?

(for STT, your action(s) could be named recognize as it is the interface method name for STT)

**Comment by GiviMAD** [comment]
I like listenAndInterpret but in the end it will do three steps (recognize -> interpret -> say) so I think it could be confusing, For me it should be listenAndAnswer or just listenNow as give a response is the normal operation.

**Comment by lolodomo** [comment]
listenAndAnswer is currently asynchronous in my implementation. It returns fast. I believe it should rather be synchronous and could even return the result of the interpreter. 
@GiviMAD : WDYT?

**Comment by lolodomo** [comment]
Off topic before I open a nee issue: were you able to use the system HLI? I have the strange feeling that it is broken as any of my commands are "rejected".

**Comment by GiviMAD** [comment]
>listenAndAnswer is currently asynchronous in my implementation. It returns fast. I believe it should rather be synchronous and could even return the result of the interpreter.

Yes I see a lot of value on make it synchronous, that way is easier to propagate any possible errors. 
About returning the results of the interpreter, maybe makes sense to also add an option to avoid STT execution so you can just get the response text without it been told?

>Off topic before I open a nee issue: were you able to use the system HLI? I have the strange feeling that it is broken as any of my commands are "rejected".

No, I was testing the new interpreter I was working on and it works correctly https://github.com/openhab/openhab-addons/pull/12260. I was able to send commands and read item states. Can you give it a try with any of the examples in the readme and tell me if it works for you?
I will try the standard interpreter in the afternoon to check if it's working for me.

**Comment by lolodomo** [comment]
> No, I was testing the new interpreter I was working on and it works correctly [openhab/openhab-addons#12260](https://github.com/openhab/openhab-addons/pull/12260). I was able to send commands and read item states. Can you give it a try with any of the examples in the readme and tell me if it works for you?

Step by step. I will try (and review) your PR in a second time.

> I will try the standard interpreter in the afternoon and let you know if it's working for me.

In fact, I think I know where is my problem. The system HLI is not able to match my item because my item is inside a group which has a similar label as the group. So the HLI is matching my text with an item and a group and finally rejects the command telling me that more than one object was found. I am going to update th elabel of my groups to be sure.

**Comment by lolodomo** [comment]
> In fact, I think I know where is my problem. The system HLI is not able to match my item because my item is inside a group which has a similar label as the item. So the HLI is matching my text with an item and a group and finally rejects the command telling me that more than one object was found. I am going to update the label of my groups to be sure.

Confirmed. By changing few group names, I was able to make the system HLI working again.
The French rules should be slightly enhanced to be usable in conjonction with Vosk. As an example, in French, it is strickly impossible for a STT to distinguish "allumer" et "allumez". Vosk converts to "allumez" while the rule in the system HLI expects "allumer". But sending a command "Allumez ..." to a voice assistant will never be used, a user will either use "Allumer ..." or "Allume ...". So, I could add "allume" to system HLI but I don't know how to handle properly allumez vs allumer.

So, at least now, I can confirm that the "listen and answer" is working with the 2 HLI already provided with openHAB.
